reconstruction/eb/src/main/java/org/jlab/service/eb/EBMatching.java

```diff
@@ -163,9 +165,90 @@ public List<DetectorParticle> findNeutrals(int ecalLayer) {
 
     /**
      * find CD neutrals and append to particle list
+	 * @param de
+	 * @return whether any neutrals were added 
      */
     public boolean addCentralNeutrals(DetectorEvent de) {
 
+        int newNeutrals = 0;
+        
+        Vector3 vertex = new Vector3(0,0,0);
+        if (!de.getParticles().isEmpty()) {
+            vertex.copy(eventBuilder.getEvent().getParticle(0).vertex());
+        }
+
+        //
+        // This is the new case where CND only does intralayer clustering,
+        // and EB does CND's interlayer matching:
+        //
+        MatchCylindrical matcher = new MatchCylindrical(
+		eventBuilder.ccdb.getDouble(EBCCDBEnum.CND_DZ),
+		eventBuilder.ccdb.getDouble(EBCCDBEnum.CND_DPHI),
+		eventBuilder.ccdb.getDouble(EBCCDBEnum.CND_DT));
+        List<DetectorResponse> cnd = eventBuilder.getUnmatchedResponses(null, DetectorType.CND, 0);
+
+        // This would be energy-based seeding, with no layer preferences:
+        //hits.sort(DetectorResponseComparators.energyCompare);
+        // This would be layer-based seeding, with energy preference within layers:
+        cnd.sort(DetectorResponseComparators.layerEnergy);
+
+	while (!cnd.isEmpty()) {
+            // CND clusters are already sorted such that the first one will
+	    // always seed a new neutral:
+            DetectorResponse seed = cnd.remove(0);
+	    final int pindex = de.getParticles().size();
+            DetectorParticle neutral = DetectorParticle.createNeutral(seed,vertex);
+            seed.setAssociation(pindex);
+            newNeutrals++;
+
+	    // Associate any remaining CND clusters to this neutral:
+            for (int i=0; i<cnd.size(); i++) {
+                if (seed.getDescriptor().getLayer() !=  cnd.get(i).getDescriptor().getLayer()) {
+		    if (matcher.matches(neutral, cnd.get(i))) {
+			DetectorResponse hit = cnd.remove(i);
+			neutral.addResponse(hit);
+			hit.setAssociation(pindex);
+			break;
```

Comment:
is this break statement imposing that from all the remaining cluster only one is matched? so maximum number of cnd matched clusters is 2?
sorry I did not understand why you break and not look for other matching clusters

Reply:
Ah, yes, thanks!   I overlooked that since so few neutrons really have multiple CND hits.  It's addressed with the next commit.